### ISSUE TITLE
Building fails from source on Fedora 28 with RTX 2080

### ISSUE THREAD

**Issue by YiruS** [colmap/colmap#585]
**Describe the bug**
The error happens when running "make ", it shows: [46%] Linking CXX static library libflann.a 
[46%] Built target flann
Then it reports error: make: *** [Makefile:130: all] Error 2

**To Reproduce**
Because my GPU is RTX 2080, I'd like to include more information:
CUDA: 9.2
gcc: 7.4.0
Operating system: Fedora 28
COLmap version: 3.5

Steps to install:
1. download  Colmap 3.5 unzip it
2. mkdir build
3. cd build
4. cmake -DCMAKE_C_COMPILER=/home/cusers/gcc/bin/gcc -DCMAKE_CXX_COMPILER=/home/cusers/gcc/bin/g++ -DCUDA_NVCC_FLAGS="-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_75,code=compute_75" .. 
5. make -j8 (where error happens)

**Screenshots**
![screenshot-error](https://user-images.githubusercontent.com/14182280/53535684-946bd880-3ab8-11e9-96f6-f8c482128da6.png)

**Comment by ahojnnes** [comment]
Potentially running out of memory on this machine? Otherwise, there should be some other error message further up.

**Comment by YiruS** [comment]
It's been fixed now. Need to manually change the CUDA_ARCH to Pascal and cmake would work.

**Comment by Zixin-YANG** [comment]
> It's been fixed now. Need to manually change the CUDA_ARCH to Pascal and cmake would work.

Hello, did you change the CUDA_ARCH? in the Cmakelist?

**Comment by lsx1980** [comment]
same issue, looking for solutions.
Ubuntu 20.04 ,
Device 0: "GeForce RTX 2070 SUPER"
CUDA Driver Version / Runtime Version 11.2 / 10.1
CUDA Capability Major/Minor version number: 7.5.

**Comment by lsx1980** [comment]
Problem solved, solution:

    check your compiler version
    dpkg --list | grep compiler
    ii g++ 4:9.3.0-1ubuntu2 amd64 GNU C++ compiler
    ii g++-8 8.4.0-3ubuntu2 amd64 GNU C++ compiler
    ii g++-9 9.3.0-17ubuntu120.04 amd64 GNU C++ compiler
    ii gcc 4:9.3.0-1ubuntu2 amd64 GNU C compiler
    ii gcc-8 8.4.0-3ubuntu2 amd64 GNU C compiler
    ii gcc-9 9.3.0-17ubuntu120.04 amd64 GNU C compiler
    git check out master branch instead of dev
    git checkout master
    Use gcc and g++ version 8 to compile
    CC=/usr/bin/gcc-8 CXX=/usr/bin/g++-8 cmake ..

It works for me now:)

Ubuntu 20.04 ,
Device 0: "GeForce RTX 2070 SUPER"
CUDA Driver Version / Runtime Version 11.2 / 10.1
CUDA Capability Major/Minor version number: 7.5

**Comment by YifuTao** [comment]
I ran into the same issue and it exits on building `flann`.
`CC=/usr/bin/gcc-8 CXX=/usr/bin/g++-8 cmake ..` works and here are some more details.

1. The error message will show up if run `make -j` again. It was cuda version incompatible with gcc. In my case I was using CUDA 10.2 which only supports upto gcc8, but I used gcc 9.
2. Solution is therefore either change CUDA or gcc version to make them compatible, and hence the above solution
